subsys/net/ip/net_if.c

```diff
@@ -125,6 +125,49 @@ static sys_slist_t timestamp_callbacks;
 #define debug_check_packet(...)
 #endif /* CONFIG_NET_IF_LOG_LEVEL >= LOG_LEVEL_DBG */
 
+struct net_if *z_impl_net_if_get_by_index(int index)
+{
+	if (index <= 0) {
+		return NULL;
+	}
+
+	if (&__net_if_start[index - 1] >= __net_if_end) {
+		NET_DBG("Index %d is too large", index);
+		return NULL;
+	}
+
+	return &__net_if_start[index - 1];
+}
+
+#ifdef CONFIG_USERSPACE
+struct net_if *z_vrfy_net_if_get_by_index(int index)
+{
+#if defined(CONFIG_NET_IF_USERSPACE_ACCESS)
```

Comment:
what are the semantics of CONFIG_NET_IF_USERSPACE_ACCESS?

No user thread has permission to access a network interface object without being granted it first, so I don't see why this is still here.

Reply:
> what are the semantics of CONFIG_NET_IF_USERSPACE_ACCESS?

To me allowing a user thread access to net_if is kind of "super user" access. So I added a separate Kconfig variable that controls whether this kind of access is possible at all, and end-user has to make a conscious effort to enable this.